### PR TITLE
Removing redundancies in System Solver HIP example

### PR DESCRIPTION
**Pull request by pelesh**
Minor changes: merge fixes by @kswirydo and @pelesh, remove redundant calls and fix comments.

### CODE REVIEW

File: examples/r_SysSolverCuda.cpp [hunk 1]
```diff
@@ -51,9 +51,6 @@ int main(int argc, char *argv[])
 
   ReSolve::SystemSolver* solver = new ReSolve::SystemSolver(workspace_CUDA);
 
-  // ReSolve::LinSolverDirectKLU* KLU = new ReSolve::LinSolverDirectKLU;
```
Comment (kswirydo) [inline examples/r_SysSolverCuda.cpp:54]: Why is this removed?

File: examples/r_SysSolverHip.cpp [hunk 2]
```diff
@@ -51,8 +51,6 @@ int main(int argc, char *argv[] )
   vector_type* vec_r;
 
   ReSolve::SystemSolver* solver = new ReSolve::SystemSolver(workspace_HIP);
-  // ReSolve::LinSolverDirectKLU* KLU = new ReSolve::LinSolverDirectKLU;
-  // ReSolve::LinSolverDirectRocSolverRf* Rf = new ReSolve::LinSolverDirectRocSolverRf(workspace_HIP);
```
Comment (kswirydo) [inline examples/r_SysSolverHip.cpp:55]: Why is this removed?

File: resolve/SystemSolver.cpp [hunk 3]
```diff
@@ -52,6 +52,7 @@ namespace ReSolve
     refactorizationMethod_ = "glu";
     solveMethod_ = "glu";
```
Comment (kswirydo) [inline resolve/SystemSolver.cpp:53]: ??

File: resolve/SystemSolver.cpp [hunk 4]
```diff
@@ -52,6 +52,7 @@ namespace ReSolve
...
     irMethod_ = "none";
+    gsMethod_ = "cgs2";
```
Comment (kswirydo) [inline resolve/SystemSolver.cpp:55]: if there is no IR, you dont need to set this option at all

File: resolve/SystemSolver.cpp [hunk 5]
```diff
@@ -84,6 +86,8 @@ namespace ReSolve
   {
     A_ = A;
     dummy_ = new vector_type(A->getNumRows());
+    dummy_->allocate(memory::DEVICE);
```
Comment (kswirydo) [inline resolve/SystemSolver.cpp:89]: do you destroy this vector anywhere?

**Review by kswirydo — APPROVED** [review]
I didnt run it yet.

File: examples/r_SysSolverCuda.cpp (hunk 1 above)
Comment (pelesh) [reply examples/r_SysSolverCuda.cpp:54]: This is instantiated within SystemSolver now.

File: examples/r_SysSolverHip.cpp (hunk 2 above)
Comment (pelesh) [reply examples/r_SysSolverHip.cpp:55]: Same here -- specific solvers are set within SystemSolver.

File: resolve/SystemSolver.cpp (hunk 4 above)
Comment (pelesh) [reply resolve/SystemSolver.cpp:55]: This is a placeholder for now.

File: resolve/SystemSolver.cpp (hunk 5 above)
Comment (pelesh) [reply resolve/SystemSolver.cpp:89]: I destroy this in the destructor, but this needs to be improved. There are corner cases where you would want to destroy system solver before you set matrix `A_`.

**Comment by kswirydo** [comment]
I run it and it run on frontier.
